tower-http/src/services/fs/serve_dir.rs

```diff
@@ -424,7 +459,7 @@ fn append_slash_on_path(uri: Uri) -> Uri {
 enum Output {
     File(FileRequest),
     Redirect(HeaderValue),
-    NotFound,
+    StatusCode(StatusCode),
```

Comment:
Feels unfortunate that this went from one possibility to hundreds when it seemingly only needed two possibilities.

Reply:
Yes, you're right!

Reply:
the `check_modified_headers` can output 2 different status codes: `304 Not Modified` and `412 Precondition Failed` this is why I'm not using 2 more enum variants.